src/naive/datetime/mod.rs

```diff
@@ -272,8 +272,10 @@ impl NaiveDateTime {
     ///            Ok(NaiveDate::from_ymd_opt(2015, 7, 1).unwrap().and_hms_milli_opt(8, 59, 59, 1_123).unwrap()));
     /// ```
     ///
-    /// Missing seconds are assumed to be zero,
-    /// but out-of-bound times or insufficient fields are errors otherwise.
+    /// Missing hours, minutes, seconds and nanoseconds are assumed to be zero.
+    /// A missing day will, depending on the other available fields, default to the first day of
+    /// the month or the first day of the week.
+    /// Out-of-bound times or insufficient fields are errors otherwise.
     ///
```

Comment:
I really like this detailed explanation in the docstring.

However, it's somewhat confusing.

My suggestion includes referring specifically to `fmt`, using the formal phase _specifier_ (as in "strftime specifier" which is how it's referred elsewhere) and being more explicit about the inferred subject:

    /// If `fmt` does not contain an hours, minutes, seconds or nanoseconds specifier than that is assumed to be zero.
    //// If `fmt` does not contain a day specifier then it will first day of the month or first day of the week, depending upon other specifiers.
    /// Otherwise, out-of-bound times or insufficient fields are errors.

Reply:
Thank you for the suggestion. I have updated it to:
```rust
    /// If `fmt` does not contain enough specifiers to get a complete date, parsing inserts a
    /// default value:
    /// - If `fmt` has year and month specifiers parsing assumes it is the first day of the month.
    /// - If `fmt` has year and week specifiers parsing assumes it is the first day of the week.
```